### PR TITLE
Convert audio data to spectrogram

### PR DESCRIPTION
**Pull request by 708yamaguchi**
~~DO NOT MERGE YET.~~
~~I am trying to create program to recognize sound.~~

I added programs to convert
`/audio` -> `/spectrum` -> `/spectrogram`

I also added sample rosbag, sample program and test program.

@k-okada

### CODE REVIEW

File: jsk_perception/audio_to_spectrogram/audio_to_spectrogram.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+import rospy
+
+from audio_common_msgs.msg import AudioData
+from sensor_msgs.msg import Image
+
+
+# This node publish spectrogram (sensor_msgs/Image)
+# from audio (audio_common_msgs/AudioData)
+# Respeaker is assumed to be used.
```
Comment (itohdak) [inline jsk_perception/audio_to_spectrogram/audio_to_spectrogram.py]: Which part depends on respeaker?

Comment (708yamaguchi) [reply jsk_perception/audio_to_spectrogram/audio_to_spectrogram.py]: Thank you for your review.

I added dependency like below:
https://github.com/jsk-ros-pkg/jsk_recognition/pull/2478/files#diff-28ff2534b886febe47348c0b55f459bfR14-R15

**Comment by 708yamaguchi** [comment]
In this program, audio topic (`audio_common_msgs/AudioData`) is converted into spectrum topic (`jsk_recognition_msgs/Spectrum`, original msg type) and spectrogram topic (`sensor_msgs/Image`).

I add sample launch file and rosbag. You can use this program by the command below:
```bash
roslaunch jsk_perception sample_audio_to_spectrogram.launch  # play rosbag and visualize audio
```

Visualization sample of 300 Hz audio:
spectrum

spectrogram

**Comment by 708yamaguchi** [comment]
I fixed some bugs and made `audio_to_spectrogram` as ROS package.

**Comment by 708yamaguchi** [comment]
Thank you for merging.
There is one thing I forgot to tell.

If you use melodic, please be aware that until https://github.com/ros-perception/image_pipeline/commit/9dd6ce31b7ce89d0a0b9a46b7427367b3726c7e1 is released, the `image_view`'s `colormap` will not work and the spectrogram in `image_view` will be shown in grayscale.

**Pull request merged**